app/src/main/java/com/dreamteam/paca/GPSTracker.java

```diff
@@ -0,0 +1,161 @@
+package com.dreamteam.paca;
+
+import android.app.AlertDialog;
+import android.app.Service;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.os.IBinder;
+import android.provider.Settings;
+
+/**
+ * Created by Dan Flanagan on 4/8/2015.
+ * This does all the framework for using the device's built in gps system
+ */
+public class GPSTracker extends Service implements LocationListener {
+
+    private final Context context;
+
+    boolean isGPSEnabled = false;
+    boolean isNetworkEnabled = false;
+    boolean canGetLocation = false;
+
+    Location location;
+
+    private double latitude;
+    private double longitude;
+
+    private static final long MIN_DISTANCE_FOR_UPDATE = 15;
+    private static final long MIN_TIME_BETWEEN_UPDATES = 1000 * 60 * 1;
+
+    protected LocationManager locationManager;
+
+    public GPSTracker(Context context) {
+        this.context = context;
+        getLocation();
+    }
+
+    public Location getLocation() {
+        try {
+            locationManager = (LocationManager) context.getSystemService(LOCATION_SERVICE);
+            isGPSEnabled = locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
+            isNetworkEnabled = locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
+
+            if (!isGPSEnabled && !isNetworkEnabled) {
+                showSettingsAlert();
+            } else {
+                this.canGetLocation = true;
+
+                if (isNetworkEnabled) {
+                    locationManager.requestLocationUpdates(LocationManager.NETWORK_PROVIDER, MIN_TIME_BETWEEN_UPDATES, MIN_DISTANCE_FOR_UPDATE, this);
+
+                    if (locationManager != null) {
+                        location = locationManager.getLastKnownLocation(LocationManager.NETWORK_PROVIDER);
+
+                        if (location != null) {
+                            latitude = location.getLatitude();
+                            longitude = location.getLongitude();
+                        }
+                    }
+                }
+                if (isGPSEnabled) {
+                    if (location == null) {
+                        locationManager.requestLocationUpdates(LocationManager.NETWORK_PROVIDER, MIN_TIME_BETWEEN_UPDATES, MIN_DISTANCE_FOR_UPDATE, this);
+
+                        if (locationManager != null) {
+                            location = locationManager.getLastKnownLocation(LocationManager.NETWORK_PROVIDER);
+                            if (location != null) {
+                                latitude = location.getLatitude();
+                                longitude = location.getLongitude();
+                            }
+                        }
+                    }
+                }
+            }
+
+        } catch (Exception e) {
```

Comment:
Don't catch generic exception. We need to know what exception we are catching here